python/cudf/cudf/tests/test_list.py

```diff
@@ -336,25 +337,67 @@ def test_concatenate_list_with_nonlist():
 
 
 @pytest.mark.parametrize(
-    "indata,expect",
+    "data",
     [
-        ([1], [1]),
-        ([1, 2, 3], [1, 2, 3]),
-        ([[1, 2, 3], [4, 5, 6]], [[1, 2, 3], [4, 5, 6]]),
-        ([None], [NA]),
-        ([1, None, 3], [1, NA, 3]),
-        ([[1, None, 3], [None, 5, 6]], [[1, NA, 3], [NA, 5, 6]]),
```

Comment:
Before https://github.com/rapidsai/cudf/pull/8442, if we wanted to make a series with an `NA` in it out of a list, we had to pass `None`. This is because the list ends up getting converted into an arrow array and then we know how to make a series or other gpu object out of that. But it didn't recognize our own `cudf.NA` object and would crash. This meant we needed to pass `None` to indicate `NA` and then after rountripping, test that there was `NA` in the same places there was `None` originally. After that PR though, we can rountrip `NA` correctly.